src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -1039,6 +1041,213 @@ class npc_magister_aledis : public CreatureScript
     }
 };
 
+enum WatchCommanderLeonus
+{
+    SAY_COVER                   = 0,
+    EVENT_LEONUS_TALK           = 1,
+    EVENT_INFERNAL_RAIN_ATTACK  = 2,
+    EVENT_FEAR_CONTROLLER_CAST  = 3,
+    EVENT_ACTIVE_FALSE          = 4,
+    NPC_INFERNAL_RAIN           = 18729,
+    SPELL_INFERNAL_RAIN         = 33814,
+    NPC_FEAR_CONTROLLER         = 19393,
+    DATA_ACTIVE                 = 1,
+};
+
+struct npc_watch_commander_leonus : public ScriptedAI
+{
+    npc_watch_commander_leonus(Creature* creature) : ScriptedAI(creature)
+    {
+        _events.ScheduleEvent(EVENT_LEONUS_TALK, 2min, 10min);
+        _events.ScheduleEvent(EVENT_INFERNAL_RAIN_ATTACK, 2min, 10min);
+        _events.ScheduleEvent(EVENT_FEAR_CONTROLLER_CAST, 2min, 10min);
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        switch (data)
+        {
+        case 1:
+            _events.ScheduleEvent(EVENT_ACTIVE_FALSE, 1s);
+            break;
+        default:
+            break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_LEONUS_TALK:
+                    Talk(SAY_COVER);
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_SHOUT);
+                    break;
+                case EVENT_INFERNAL_RAIN_ATTACK:
+                {
+                    std::list<Creature*> infernalrainList;
+                    Trinity::AllCreaturesOfEntryInRange checkerInfernalrain(me, NPC_INFERNAL_RAIN, 200.0f);
+                    Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher_infernal(me, infernalrainList, checkerInfernalrain);
+                    Cell::VisitAllObjects(me, checkerInfernalrain, 200.0f);
+
+                    if (infernalrainList.empty())
+                        return;
+
+                    for (Creature* infernal : infernalrainList)
+                        if (!infernal->isMoving() && infernal->GetPositionZ() > 118.0f)
+                            infernal->AI()->SetData(DATA_ACTIVE, DATA_ACTIVE);
+
+                    break;
+                }
+                case EVENT_FEAR_CONTROLLER_CAST:
+                {
+                    std::list<Creature*> fearcontrollerList;
+                    Trinity::AllCreaturesOfEntryInRange checkerFear(me, NPC_FEAR_CONTROLLER, 200.0f);
+                    Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher_fear(me, fearcontrollerList, checkerFear);
+                    Cell::VisitAllObjects(me, checkerFear, 200.0f);
+
+                    if (fearcontrollerList.empty())
+                        return;
```

Comment:
do you want to return here or just skip the list stuff? currently you skip UpdateVictim and melee attack (also empty checks are pointless before regular loops)